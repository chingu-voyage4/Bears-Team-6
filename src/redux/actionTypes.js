// @flow

import { Types as timestamp } from './timestamp/actions'
import { Types as websocketProvider } from './websocketProvider/actions'
import { Types as auth } from './auth/actions'

export const ActionTypes = {
  ...websocketProvider,
  ...timestamp,
  ...auth,
}
