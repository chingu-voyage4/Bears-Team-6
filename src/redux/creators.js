// @flow

import { Creators as timestamp } from './timestamp/actions'
import { Creators as websocketProvider } from './websocketProvider/actions'

export const Creators = {
  ...websocketProvider,
  ...timestamp,
}
