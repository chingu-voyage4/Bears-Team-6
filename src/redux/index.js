// @flow
export { Creators } from './creators'
export { ActionTypes } from './actionTypes'
export { reducers } from './reducers'
export { sagas } from './sagas'
