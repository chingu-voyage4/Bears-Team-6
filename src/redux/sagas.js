// @flow
import { all } from 'redux-saga/effects'
import type { Saga } from '../types'
import { timestamp } from './timestamp/saga'
import { websocketProvider } from './websocketProvider/saga'
import { auth } from './auth/saga'
import { userSettings } from './userSettings/saga'

const allSagas = [timestamp, websocketProvider, auth, userSettings]

export function* sagas(): Saga<void> {
  yield all(allSagas.reduce((xs, x) => xs.concat(x), []))
}
