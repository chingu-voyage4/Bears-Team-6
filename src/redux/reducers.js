// @flow

import { combineReducers } from 'redux'
import { timestamp } from './timestamp/reducer'

export const reducers = combineReducers({
  timestamp,
})
