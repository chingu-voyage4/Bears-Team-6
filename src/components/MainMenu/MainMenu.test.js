// @flow

import React from 'react'
import { shallow } from 'enzyme'
import { createMockStore } from 'redux-test-utils'
import { MainMenu } from '.'
import { reducers } from '../../redux'

const initialReduxState = reducers({}, {})
const store = createMockStore(initialReduxState)

describe('MainMenu', () => {
  const wrapper = shallow(<MainMenu />, { context: { store } })
  it('renders as expected', () => {
    expect(wrapper.dive()).toMatchSnapshot()
  })
})
