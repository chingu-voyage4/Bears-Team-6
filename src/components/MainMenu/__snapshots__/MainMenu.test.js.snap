// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenu renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainMenu
    dispatch={[Function]}
    isAuthenticated={false}
    isLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <nav>
        <ul>
          <li>
            <Link
              replace={false}
              to="/registration"
            >
              Registration
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
          </li>
        </ul>
      </nav>,
      "className": Object {
        "display": "flex",
        "justifyContent": "space-between",
        "listStyle": "none",
        "width": 300,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
            <Link
              replace={false}
              to="/registration"
            >
              Registration
            </Link>
          </li>
          <li>
            <Link
              replace={false}
              to="/login"
            >
              Login
            </Link>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                replace={false}
                to="/registration"
              >
                Registration
              </Link>
            </li>,
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "/registration",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/registration"
              >
                Registration
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Registration",
                "replace": false,
                "to": "/registration",
              },
              "ref": null,
              "rendered": "Registration",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "/login",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Login",
                "replace": false,
                "to": "/login",
              },
              "ref": null,
              "rendered": "Login",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <nav>
          <ul>
            <li>
              <Link
                replace={false}
                to="/registration"
              >
                Registration
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>
            </li>
          </ul>
        </nav>,
        "className": Object {
          "display": "flex",
          "justifyContent": "space-between",
          "listStyle": "none",
          "width": 300,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <ul>
            <li>
              <Link
                replace={false}
                to="/registration"
              >
                Registration
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/login"
              >
                Login
              </Link>
            </li>
          </ul>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <Link
                  replace={false}
                  to="/registration"
                >
                  Registration
                </Link>
              </li>,
              <li>
                <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "/registration",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/registration"
                >
                  Registration
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Registration",
                  "replace": false,
                  "to": "/registration",
                },
                "ref": null,
                "rendered": "Registration",
                "type": [Function],
              },
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "/login",
              "nodeType": "host",
              "props": Object {
                "children": <Link
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Login",
                  "replace": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": "Login",
                "type": [Function],
              },
              "type": "li",
            },
          ],
          "type": "ul",
        },
        "type": "nav",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
